### PR TITLE
Support redis-rb 5.x

### PR DESCRIPTION
**Pull request by hieuk09**
Some notable changes:

- Update integration test to use real redis client & server instead of `MockRedis` in order to test correct behavior of redis-rb
- Add test to ensure the gem work with both redis-rb 4.x & 5.x
- Update some parts to handle errors when upgrading to redis-rb 5.0

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -9,11 +9,21 @@ jobs:
...
           - "3.0"
           - "2.7"
+        redis:
+          - 4
+          - 5
+        redis-version:
+          - 5
     env:
       RUBY_VERSION: ${{ matrix.ruby }}
+      BUNDLE_GEMFILE: Gemfile.redis.${{ redis }}
```
Comment (kennethteh90) [inline .github/workflows/main.yml]: Does this need to be `matrix.redis`?

Comment (hieuk09) [reply .github/workflows/main.yml]: Oh right, my bad 🤦

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 require 'heavy_keeper'
 require 'mock_redis'
+require 'redis'
+require 'byebug'
```
Comment (kennethteh90) [inline spec/spec_helper.rb:6]: Hm! Is this line needed? I would have thought it would require automatically.

**Review by kennethteh90 — APPROVED** [review]

File: spec/spec_helper.rb (hunk 2 above)
Comment (hieuk09) [reply spec/spec_helper.rb:6]: Yes, we need it. It's only required automatically if we use `require 'bundler/setup'`

Comment (kennethteh90) [reply spec/spec_helper.rb:6]: Ohhh. TIL! 🙏

**Pull request merged**